src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -1,108 +1,918 @@
+
+            POLY_OPA_DISP = func_8010DC58(POLY_OPA_DISP, sCloudTextures[n], D_8082B7F0[n], D_8082B838[n], 0);
+        }
+    }
+
+    if ((pauseCtx->state >= 0x15) && (pauseCtx->state <= 0x19)) {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetRenderMode(POLY_OPA_DISP++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+        gDPSetCombineMode(POLY_OPA_DISP++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, R_PAUSE_OWLWARP_ALPHA);
+        gDPFillRectangle(POLY_OPA_DISP++, 50, 62, 270, 190);
+    }
+
+    func_8012C8AC(play->state.gfxCtx);
+
+    if (!((pauseCtx->state >= 0x15) && (pauseCtx->state <= 0x19))) {
+        gDPLoadTextureBlock(POLY_OPA_DISP++, gWorldMapDotTex, G_IM_FMT_IA, G_IM_SIZ_8b, 8, 8, 0,
+                            G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD,
+                            G_TX_NOLOD);
+        gDPSetCombineLERP(POLY_OPA_DISP++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0,
+                          PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, D_8082B584[0], D_8082B584[1], D_8082B584[2], pauseCtx->alpha);
+
+        gDPSetEnvColor(POLY_OPA_DISP++, D_8082B590[0], D_8082B590[1], D_8082B590[2], 0);
+
+        if (R_PAUSE_DBG_MAP_CLOUD_ON) {
+            gSaveContext.save.worldMapCloudVisibility |= (u16)~0x8000;
+
+            pauseCtx->mapPageVtx[120].v.ob[0] = pauseCtx->mapPageVtx[122].v.ob[0] = R_PAUSE_DBG_MAP_CLOUD_X;
+
+            pauseCtx->mapPageVtx[121].v.ob[0] = pauseCtx->mapPageVtx[123].v.ob[0] =
+                pauseCtx->mapPageVtx[120].v.ob[0] + 8;
+
+            pauseCtx->mapPageVtx[120].v.ob[1] = pauseCtx->mapPageVtx[121].v.ob[1] = R_PAUSE_DBG_MAP_CLOUD_Y;
+
+            pauseCtx->mapPageVtx[122].v.ob[1] = pauseCtx->mapPageVtx[123].v.ob[1] =
+                pauseCtx->mapPageVtx[120].v.ob[1] - 8;
+        }
+
+        // Loop over RegionId (i), unused vtxIndex (j), unused (k)
+        for (i = 0, j = 0; i < REGION_MAX; i++, k++, j += 4) {
+            if (pauseCtx->worldMapPoints[i]) {
+                gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[120 + i * 4], 4, 0);
+                gSP1Quadrangle(POLY_OPA_DISP++, 0, 2, 3, 1, 0);
+            }
+        }
+
+    } else {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetCombineMode(POLY_OPA_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+        gDPLoadTextureBlock(POLY_OPA_DISP++, gWorldMapOwlFaceTex, G_IM_FMT_RGBA, G_IM_SIZ_32b, 24, 12, 0,
+                            G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD,
+                            G_TX_NOLOD);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+
+        if (R_PAUSE_DBG_MAP_CLOUD_ON) {
+            gSaveContext.save.worldMapCloudVisibility |= (u16)~0x8000;
```

Comment:
Hmm, well that's obviously not what was written originally unless they used the top half for something else.